src/icon.rs

```diff
@@ -21,10 +22,20 @@ const ICON_SPACE: &str = "  ";
 // s#\\u[0-9a-f]*#\=eval('"'.submatch(0).'"')#
 impl Icons {
     pub fn new(theme: Theme) -> Self {
+        let display_icons = theme == Theme::Default || theme == Theme::Unicode;
+        let (icons_by_name, icons_by_extension) = if theme == Theme::Default {
+            (
+                Self::get_default_icons_by_name(),
+                Self::get_default_icons_by_extension(),
+            )
+        } else {
+            (HashMap::new(), HashMap::new())
```

Comment:
Here you need to call two new functions `get_unicode_icons_by_name()` and `get_default_icons_by_extension` which will generate two HashMap with the corresponding unicode icons.

Reply:
The idea was that if the maps are empty, every call to `get` will fall through to the default case.

Reply:
What would you say to making the icon set a trait with a single method
```rust
fn get(&self, name: &Name) -> String
```
The unicode icon set would always return the default file or directory icon, the “fancy” icon set would look the icons up in a table. Is that overcomplicating it?